### PR TITLE
Update variables.md to elaborate on variable syntax usage.

### PR DESCRIPTION
**Pull request by JFolberth**
This pull request includes updates to the documentation in `docs/pipelines/process/variables.md` to improve clarity on variable usage in Azure DevOps pipelines. The most important changes focus on refining the guidance for using macro and template syntax.

Documentation updates:

* Clarified that macro syntax should be used for providing secure strings or predefined variable inputs for a task.
* Expanded the explanation on template variables, emphasizing their use for fully injecting variable values at pipeline compilation and noting that template syntax should not be used for sensitive values.

### CODE REVIEW

**Comment by Court72** [comment]
@juliakm

Can you review the proposed changes? 
	
**Important**: When the changes are ready for publication, adding a `#sign-off` comment is the best way to signal that the PR is ready for the review team to merge. 
	
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by JFolberth** [comment]
@juliakm Are you able to review?

File: docs/pipelines/process/variables.md [hunk 1]
```diff
@@ -146,11 +146,11 @@ Runtime expression variables are only expanded when they're used for a value, no
 
 ### What syntax should I use?
 
-Use macro syntax if you're providing input for a task. 
+Use macro syntax if you're providing a secure string or a [predefined variable](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml) input for a task.
```
Comment (juliakm) [inline docs/pipelines/process/variables.md:149]:
```suggestion
Use macro syntax if you're providing a secure string or a [predefined variable](/azure/devops/pipelines/build/variables) input for a task. 
```

**Review by juliakm — CHANGES_REQUESTED** [review]
Thanks for your contribution! Can you make a couple of small changes?

**Comment by JFolberth** [comment]
@juliakm Updates were made as requested.

**Review by juliakm — APPROVED** [review]
Thanks for the updates!

**Comment by juliakm** [comment]
#sign-off

**Pull request merged**